code-challenges.test.js

```diff
@@ -10,43 +10,194 @@
+  return arr.filter((element) => element.toLowerCase().includes(query.toLowerCase()))
+}
+
+// PASS  ./code-challenges.test.js
+// wordsWithAorB
+// ✓ takes in an array of words and a single letter and returns an array of all the words containing that particular letter. (1 ms)
+
+//------------------------------------------------------------------------------------------------------------
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("fullHouseCheck", () => {
+  it("takes in an array of 5 numbers and determines whether or not the array is a full house.", () => {
+    const hand1 = [5, 5, 5, 3, 3]
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4]
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4]
+    // Expected output: false
+    const hand4 = [7, 2, 7, 2, 7]
+    // Expected output: true
+    expect(fullHouseCheck(hand1)).toEqual(true)
+    expect(fullHouseCheck(hand2)).toEqual(false)
+    expect(fullHouseCheck(hand3)).toEqual(false)
+    expect(fullHouseCheck(hand4)).toEqual(true)
+  })
+})
+
+//FAIL  ReferenceError: fullHouseCheck is not defined
 
 // b) Create the function that makes the test pass.
+
+// I create a function and name it fullHouseCheck. 
+// I pass one parameter -> array
+// I create a variable count1 where I will store data of how many times the first element of array repeats. 
+// I set up 1 as a value of count1 variable because the first element will appear in the array at least once.
+//In addition I create a variable -> count2. I will use it to count elements for the second part of the full house combination. And the value of the variable is 0 because we have not started counting of the element. 
+//I create a variable first_value to store the first element of my array. 
+//in second_value I will store values of the element which are not equal to the first value.
+// I use variable called result to return function result -> true or false 
+// Then use for loop to iterate though my array starting from the second element of the array. 
+// If array element = to first value the count1 will be incremented by 1.
+//If array element is not = to the first element I assign it to the second_value variable 
+// I create a second for loop to calculate how many times the second value repeats in my array. 
+//After that I  create a conditional statement where I check if we have one pair and one three of a kind.
+//return true if it is a full house otherwise return false.
+
+
+
+const fullHouseCheck = (arr) => {
```

Comment:
Excellent work in this section!  This problem is super tricky, but you crushed it and your pseudocode looks excellent 🤩.  If you want to stretch yourself even further, see if you can refactor this to use higher-order functions instead of for loops.